src/App/App.tsx

```diff
@@ -1,10 +1,19 @@
+import React from 'react';
 import './App.css';
 import ToDoHeader from '../ToDoHeader';
+import ToDoList from '../ToDoList';
+import ToDoDetail from '../ToDoDetail';
 
 const App = () => {
+  const [selectedTask, setSelectedTask] = React.useState(null);
```

Comment:
```suggestion
  const [selectedTask, setSelectedTask] = React.useState();
```

🗣️ can leave this as undefined as well if you're not planning to initialize it right away.

Reply:
Whether I use `null` or `undefined`, I get a type error, `Type 'undefined' is not assignable to type '{ title: string; description: string; deadline?: Date | undefined; }'. `, unless I include a `| undefined` or `| null` in the type below.  Is there a better way of allowing `selectedTask` to be undefined?

```
type ToDoDetailProps = {
  isNewTask?: boolean,
  selectedTask: {
    title: string,
    description: string,
    deadline?: Date
  } | undefined
} & React.ComponentProps<'div'>;
```